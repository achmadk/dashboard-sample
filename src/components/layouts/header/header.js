import React, { createRef, PureComponent } from 'react'
import { connect } from 'react-redux'
import classnames from 'classnames'

import { Link } from 'react-router-dom'
import NavLeftList from './nav-left-list'
import NavRightList from './nav-right-list'

import { BRAND } from '@/constants/app-config'
import { LINK } from '@/constants/demo-data'

export class Header extends PureComponent {
  constructor (props) {
    super(props)
    this.sidebarBtn = createRef()
  }
  componentDidMount () {
    $(this.sidebarBtn.current).on('click', this.openSidebarMobile)
    // const sidebarBtn = this.sidebarBtn
    // const $sidebarToggler = $(sidebarBtn)
    // const $body = $('#body')

    // $sidebarToggler.on('click', (e) => {
    //   // _sidebar.scss, _page-container.scss
    //   $body.toggleClass('sidebar-mobile-open')
    // })
  }

  openSidebarMobile = () => {
    $('body').toggleClass('sidebar-mobile-open')
  }

  render () {
    // const { colorOption } = this.props
    return (
      <section className='app-header'>
        <div
          className={classnames('app-header-inner', {
            // 'bg-color-light': ['11', '12', '13', '14', '15', '16', '21'].indexOf(colorOption) >= 0,
            // 'bg-color-dark': colorOption === '31',
            // 'bg-color-primary': ['22', '32'].indexOf(colorOption) >= 0,
            // 'bg-color-success': ['23', '33'].indexOf(colorOption) >= 0,
            // 'bg-color-info': ['24', '34'].indexOf(colorOption) >= 0,
            // 'bg-color-warning': ['25', '35'].indexOf(colorOption) >= 0,
            // 'bg-color-danger': ['26', '36'].indexOf(colorOption) >= 0 })}
          })}
        >
          <div className='d-lg-none d-xl-none float-left'>
            <a
              href={LINK}
              className='header-icon toggle-sidebar-btn'
              ref={this.sidebarBtn}
            >
              <i className='material-icons' style={{ color: 'black' }}>
                menu
              </i>
            </a>
          </div>

          <div className='brand d-none d-lg-inline-block d-xl-inline-block'>
            <h2>
              <Link to='/'>{BRAND}</Link>
            </h2>
          </div>

          <div className='top-nav-left d-none d-lg-inline-block d-xl-inline-block'>
            <NavLeftList />
          </div>

          <div className='top-nav-right'>
            <NavRightList />
          </div>
        </div>
      </section>
    )
  }
}

// const mapStateToProps = state => ({
//   colorOption: state.settings.colorOption
// })

export default connect()(Header)
//   mapStateToProps
