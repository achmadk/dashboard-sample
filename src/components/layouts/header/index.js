import React, { lazy, Suspense } from 'react'

export const headerComponent = () =>
  import(/* webpackChunkName: "header" */ './header')

const Header = lazy(headerComponent)

export default function HeaderContainer () {
  return (
    <Suspense fallback={<div>Loading...</div>}>
      <Header />
    </Suspense>
  )
}
