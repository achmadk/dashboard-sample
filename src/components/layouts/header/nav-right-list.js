import React, { PureComponent } from 'react'
import { withRouter } from 'react-router-dom'
import { compose } from 'redux'
import { connect } from 'react-redux'

import Avatar from '@material-ui/core/Avatar'
import IconButton from '@material-ui/core/IconButton'
import Menu from '@material-ui/core/Menu'
import MenuItem from '@material-ui/core/MenuItem'
import MaterialIcon from '@/components/material-icon'

import { getUserParams } from '@/shared-state/map-state-to-props/user'

import { USER } from '@/constants/demo-data'

// import avatarImage from '@/assets/images-demo/g1.jpg'

const avatarImage = require('@/assets/images-demo/g1.jpg')

class NavRightList extends PureComponent {
  state = {
    anchorEl: null
  }

  handleClick = event => {
    // console.log( event)
    this.setState({ anchorEl: event.currentTarget })
  }

  handleClose = () => {
    this.setState({ anchorEl: null })
  }

  onLogutButtonClicked = async () => {
    const { dispatch, userParams } = this.props
    try {
      const {
        doLogout
      } = await import(/* webpackPrefetch: true, webpackChunkName: "do-logout" */
      '@/shared-state/actions/user')
      await dispatch(doLogout(userParams))
    } catch (error) {
      console.log(error)
    }
  }

  render () {
    const { anchorEl } = this.state

    return (
      <ul className='list-unstyled float-right'>
        <li className='list-inline-item search-box seach-box-right d-none d-md-inline-block'>
          <div className='search-box-inner'>
            <div className='search-box-icon'>
              <MaterialIcon icon='search' />
            </div>
            <input type='text' placeholder='search...' />
            <span className='input-bar' />
          </div>
        </li>
        <li style={{ marginRight: '10px' }}>
          <IconButton
            className='header-btn'
            aria-owns={anchorEl ? 'app-header-menu' : null}
            aria-haspopup='true'
            onClick={this.handleClick}
          >
            <Avatar
              alt='avatar'
              src={avatarImage}
              className='rounded-circle header-avatar'
            />
          </IconButton>

          <Menu
            id='app-header-menu'
            className='app-header-dropdown'
            anchorEl={anchorEl}
            open={Boolean(anchorEl)}
            onClose={this.handleClose}
          >
            <MenuItem onClick={this.handleClose}>
              {' '}
              <div>
                <span>Signed in as</span>
                <strong>{USER}</strong>
              </div>{' '}
            </MenuItem>
            <div className='divider divider-solid my-1' />
            <MenuItem onClick={this.handleClose}>
              {' '}
              <a href='#/app/dashboard'>
                {' '}
                <i className='material-icons'>
                  home
                </i> <span>Dashboard</span>{' '}
              </a>{' '}
            </MenuItem>
            <MenuItem onClick={this.handleClose}>
              {' '}
              <a href='#/app/page/about'>
                {' '}
                <i className='material-icons'>person</i> <span>About</span>{' '}
              </a>{' '}
            </MenuItem>
            <MenuItem onClick={this.handleClose}>
              {' '}
              <a href='#/app/page/services'>
                {' '}
                <i className='material-icons'>help</i> <span>Need Help?</span>{' '}
              </a>{' '}
            </MenuItem>
            <div className='divider divider-solid my-1' />
            <MenuItem onClick={this.onLogutButtonClicked}>
              {' '}
              <a>
                {' '}
                <i className='material-icons'>forward</i> <span>Log Out</span>{' '}
              </a>{' '}
            </MenuItem>
          </Menu>
        </li>
      </ul>
    )
  }
}

const mapStateToProps = ({ user }) => getUserParams(user)

export default compose(
  connect(mapStateToProps),
  withRouter
)(NavRightList)
