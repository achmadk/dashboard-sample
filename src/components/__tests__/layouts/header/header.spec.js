import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import chai from 'chai'

import { Header } from '@/components/layouts/header/header'

describe('header Layout Component', () => {
  it('render correctly', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('call openSidebarMobile method correctly', () => {
    const wrapper = shallow(<Header />)
    const result = wrapper.instance().openSidebarMobile()
    chai.expect(result).to.be.undefined
  })
})
