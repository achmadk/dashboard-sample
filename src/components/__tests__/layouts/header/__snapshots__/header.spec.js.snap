// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header Layout Component render correctly 1`] = `
<section
  className="app-header"
>
  <div
    className="app-header-inner"
  >
    <div
      className="d-lg-none d-xl-none float-left"
    >
      <a
        className="header-icon toggle-sidebar-btn"
        href="javascript:;"
      >
        <i
          className="material-icons"
          style={
            Object {
              "color": "black",
            }
          }
        >
          menu
        </i>
      </a>
    </div>
    <div
      className="brand d-none d-lg-inline-block d-xl-inline-block"
    >
      <h2>
        <Link
          to="/"
        >
          ProcurA
        </Link>
      </h2>
    </div>
    <div
      className="top-nav-left d-none d-lg-inline-block d-xl-inline-block"
    >
      <withRouter(NavLeftList) />
    </div>
    <div
      className="top-nav-right"
    >
      <ConnectFunction />
    </div>
  </div>
</section>
`;
